beacon-chain/blockchain/forkchoice/service.go

```diff
@@ -0,0 +1,460 @@
+package forkchoice
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db/filters"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// Store represents a service struct that handles the forkchoice
+// logic of managing the full PoS beacon chain.
+type Store struct {
+	ctx                context.Context
+	cancel             context.CancelFunc
+	time               uint64
```

Comment:
what is time?

Reply:
time tracks when last time we have received a new `block` or new `attestation` for blockchain service